### PR TITLE
Fix #5011.

### LINKED ISSUE #5011: literally doesn't work in overload_method
## Reporting a bug

- [ ] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [ x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

```python
import numba
from numba.extending import (overload_method, overload, models, register_model, intrinsic)
from numba.special import literally
from numba.typing import signature
from numba import cgutils

class Dummy:
    def lit(self, a):
        return a

class DummyType(numba.types.Type):
    def __init__(self):
        super().__init__(name="dummy")

@register_model(DummyType)
class DummyTypeModel(models.StructModel):
    def __init__(self, dmm, fe_type):
        members = []
        super().__init__(dmm, fe_type, members)

@intrinsic
def init_dummy(typingctx):
    def codegen(context, builder, signature, args):
        dummy = cgutils.create_struct_proxy(
            signature.return_type)(context, builder)

        return dummy._getvalue()

    sig = signature(DummyType())
    return sig, codegen

@overload(Dummy)
def dummy_overload():
    def ctor():
        return init_dummy()

    return ctor

@overload_method(DummyType, 'lit')
def lit_overload(self, a):
    def impl(self, a):
        return literally(a)
        # return a # <- work fine if do simple return

    return impl

def test_impl(a):
    d = Dummy()

    return d.lit(a)

jtest = numba.njit(test_impl)
print(jtest(5))
```
Output:
```
======================================================================
ERROR: test_literally_with_overload_method (tests.test_hpat_jit.TestHpatJitIssues)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/data/hpat_fork/hpat_2/sdc/tests/test_hpat_jit.py", line 492, in test_literally_with_overload_method
    print(jtest(5))
  File "/mnt/data/numba_git/numba_master_master/numba/dispatcher.py", line 401, in _compile_for_args
    error_rewrite(e, 'typing')
  File "/mnt/data/numba_git/numba_master_master/numba/dispatcher.py", line 344, in error_rewrite
    reraise(type(e), e, None)
  File "/mnt/data/numba_git/numba_master_master/numba/six.py", line 668, in reraise
    raise value.with_traceback(tb)
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Internal error at <numba.typeinfer.CallConstraint object at 0x7f1724938da0>.
missing a required argument: 'a'
[1] During: resolving callee type: BoundFunction((<class 'tests.test_hpat_jit.TestHpatJitIssues.test_literally_with_overload_method.<locals>.DummyType'>, 'lit') for dummy)
[2] During: typing of call at /mnt/data/hpat_fork/hpat_2/sdc/tests/test_hpat_jit.py (488)

Enable logging at debug level for details.

File "tests/test_hpat_jit.py", line 488:
        def test_impl(a):
            <source elided>

            return d.lit(a)
            ^
```

Please note, works fine with simple `overload`:
```python
import numba
from numba.extending import overload
from numba.special import literally


def lit(a):
    return a

@overload(lit)
def lit_overload(a):
    def impl(a):
        return literally(a)

    return impl

def test_impl(a):
    return lit(a)

jtest = numba.njit(test_impl)
print(jtest(5))
```
Output:
```
5
```

numba version: 6d7cddb9084d843e12bcdf9c443bc9cce56aa02f (Start cycle for 0.48.0)

### PR DESCRIPTION
**Pull request by sklam**
Fixes #5011

Fix a problem with folding arguments on methods.

### CODE REVIEW

File: numba/tests/test_literal_dispatch.py [hunk 1]
```diff
@@ -326,5 +330,72 @@ def foo(a, b):
...
+            return sig, codegen
+
+        @overload(Dummy)
+        def dummy_overload():
+            def ctor():
+                return init_dummy()
+
+            return ctor
+
+        return (DummyType, Dummy)
```
Comment (stuartarchibald) [inline numba/tests/test_literal_dispatch.py]: not for now, but someday this would be really useful as a utility

**Comment by stuartarchibald** [comment]
I think Python 2.7 is unhappy https://dev.azure.com/numba/numba/_build/results?buildId=3754&view=logs&j=3438b073-af3a-5461-9413-d4f2f10db44f&t=951b3b8f-2be8-5155-9663-f97a4245ecfe&l=291.

File: numba/tests/test_literal_dispatch.py (hunk 1 above)
Comment (sklam) [reply numba/tests/test_literal_dispatch.py]: I was thinking about the same thing. There ought to be a way to make these single-serving custom types.

**Comment by seibert** [comment]
internal CI: `numba_smoketest_cpu_7`

**Comment by seibert** [comment]
This has passed enough internal CI to merge.  (not going to wait for ARM to finish when this is not really ARM specific.)

**Pull request merged**